src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -1849,36 +1861,41 @@ public static boolean useSpecificLocationForLocalSystemData()
      *
      * @return the locations where should be stored the local system keyspaces data
      */
-    public static String[] getLocalSystemKeyspacesDataFileLocations()
+    public static File[] getLocalSystemKeyspacesDataFileLocations()
     {
         if (useSpecificLocationForLocalSystemData())
-            return new String[] {conf.local_system_data_file_directory};
+            return new File[] {localSystemDataFileDirectory};
 
-        return conf.data_file_directories.length == 0  ? conf.data_file_directories
-                                                       : new String[] {conf.data_file_directories[0]};
+        return dataDirectories.length == 0  ? dataDirectories : new File[] {dataDirectories[0]};
     }
 
     /**
      * Returns the locations where the non local system keyspaces data should be stored.
      *
      * @return the locations where the non local system keyspaces data should be stored.
      */
-    public static String[] getNonLocalSystemKeyspacesDataFileLocations()
+    public static File[] getNonLocalSystemKeyspacesDataFileLocations()
     {
-        return conf.data_file_directories;
+        return dataDirectories;
     }
 
     /**
      * Returns the list of all the directories where the data files can be stored (for local system and non local system keyspaces).
      *
      * @return the list of all the directories where the data files can be stored.
      */
-    public static String[] getAllDataFileLocations()
+    public static File[] getAllDataFileLocations()
     {
-        if (conf.local_system_data_file_directory == null)
-            return conf.data_file_directories;
+        if (!useSpecificLocationForLocalSystemData())
+            return dataDirectories;
 
-        return ArrayUtils.addFirst(conf.data_file_directories, conf.local_system_data_file_directory);
+        return ArrayUtils.addFirst(dataDirectories, localSystemDataFileDirectory);
+    }
+
+    @VisibleForTesting
+    public static void setAllDataFileLocation(File[] allDataFileLocation)
```

Comment:
This method does not feel consistent - I think it should be named `setDataDirectories` because if `useAllDataLocations()` is `true`, `getAllDataFileLocations` would always different collection than the collection passed to `setAllDataFileLocations`

Reply:
renamed to `setDataDirectories`